sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -496,6 +499,57 @@ public FileReadResponse readWithResponse(OutputStream stream, FileRange range, D
         }, logger);
     }
 
+    /**
+     * Opens a file input stream to download the file.
+     * <p>
+     *
+     * @return An <code>InputStream</code> object that represents the stream to use for reading from the file.
+     * @throws DataLakeStorageException If a storage service error occurred.
+     */
+    public final DataLakeFileInputStream openInputStream() {
```

Comment:
- why `final`? that'll make this not mockable. @alzimmermsft do we have analyzer to catch that (I know we have some in dotnet for the same)
- Should we return `InputStream` ? Is there anything in `DataLakeFileInputStream` that customer would care about? If not then we may consider moving it to `implementation` package.

Reply:
Final matches blobs. We can remove from both if necessary.

This has a getProperties method on it so users can get those on download instead of making an extra service call for that.

Reply:
I think we want to return DatalakeFileInputStream since we want to also return the PathProperties to the user. 

This came from a customer feature request for BlobInputStream

Reply:
sounds good, let's remove final

Reply:
Generally, the Java SDKs use `final` as the default for classes as once a class has been released as non-`final` there is no going back without it being a breaking change. There is common tooling in the Java ecosystem to get around mocking final classes.

Reply:
this is a method isn't it ?

Reply:
Yes, I realized this is a method now. The concept is still the same but we generally don't mark methods as `final` unless absolutely necessary.